### PR TITLE
Added Logout API.

### PR DESCRIPTION
**Pull request by shekhardahore**
1. Added logout API
2. Notification providers are now expressed in an enum.
4. Notification `add` method renamed to `subscribe`.
3. LiCoreSDKConstants renamed to LiCoreConstants.
4. LiCoreConstants updated with new values.
5. Notification token generation and request added.

### CODE REVIEW

File: Sources/LithiumExample/AppDelegate.swift [hunk 1]
```diff
@@ -26,28 +27,41 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
-        // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
-        // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
-    }
-
-    func applicationDidEnterBackground(_ application: UIApplication) {
-        // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
-        // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
+    func getNotificationSettings() {
+        UNUserNotificationCenter.current().getNotificationSettings { (settings) in
+            print("Notification settings: \(settings)")
```
Comment (Aatish11) [inline Sources/LithiumExample/AppDelegate.swift]: you can remove the print?

File: Sources/LithiumExample/AppDelegate.swift [hunk 2]
```diff
@@ -26,28 +27,41 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+                     didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        let tokenParts = deviceToken.map { data -> String in
+            return String(format: "%02.2hhx", data)
+        }
+        
+        let token = tokenParts.joined()
+        if UserDefaults.standard.string(forKey: "deviceToken") == nil {
+            UserDefaults.standard.set(token, forKey: "deviceToken")
+        }
+        print("Device Token: \(token)")
```
Comment (Aatish11) [inline Sources/LithiumExample/AppDelegate.swift]: here too

File: Sources/LithiumExample/ViewController.swift [hunk 3]
```diff
@@ -25,8 +25,15 @@ class ViewController: UIViewController {
         // Dispose of any resources that can be recreated.
     }
     @IBAction func onStartCommunity(_ sender: UIButton) {
+        let deviceToken = UserDefaults.standard.string(forKey: "deviceToken")
```
Comment (Aatish11) [inline Sources/LithiumExample/ViewController.swift]: you'd probably want to make this a constant.

Comment (shekhardahore) [reply Sources/LithiumExample/ViewController.swift]: This is just for the demo. Won't be using it anywhere else.

**Pull request merged**